File: src/main/java/org/jukeboxmc/event/entity/EntityDamageEvent.java
package org.jukeboxmc.event.entity;

import lombok.ToString;
import org.jukeboxmc.entity.Entity;
import org.jukeboxmc.event.Cancellable;

/**
 * @author LucGamesYT
 * @version 1.0
 */
@ToString
public class EntityDamageEvent extends EntityEvent implements Cancellable {

    private float damage;
    private float finalDamage;
    private final DamageSource damageSource;

    public EntityDamageEvent( Entity entity, float damage, DamageSource damageSource ) {
        super( entity );
        this.damage = damage;
        this.damageSource = damageSource;
    }

    public float getDamage() {
        return this.damage;
    }

    public void setDamage( float damage ) {
        this.damage = damage;
    }

    public float getFinalDamage() {
        return this.finalDamage;
    }

    public void setFinalDamage( float finalDamage ) {
        this.finalDamage = finalDamage;
    }

    public DamageSource getDamageSource() {
        return this.damageSource;
    }

    public enum DamageSource {

        ENTITY_ATTACK,
        FALL,
        VOID,
        PROJECTILE,
        DROWNING,
        CACTUS,
        LAVA,
        ON_FIRE,
        FIRE,
        ENTITY_EXPLODE,
        MAGIC_EFFECT,
        STARVE,
        API,
        COMMAND,

    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/entity/EntityDamageEvent.kt
package org.jukeboxmc.api.event.entity

import org.jukeboxmc.api.entity.Entity
import org.jukeboxmc.api.event.Cancellable

open class EntityDamageEvent(
    entity: Entity,
    private var damage: Float,
    private val damageSource: DamageSource) : EntityEvent(entity), Cancellable {

    private var finalDamage: Float = 0F

    fun getDamage(): Float {
        return this.damage
    }

    fun setDamage(damage: Float) {
        this.damage = damage
    }

    fun getFinalDamage(): Float {
        return this.finalDamage
    }

    fun setFinalDamage(finalDamage: Float) {
        this.finalDamage = finalDamage
    }

    fun getDamageSource(): DamageSource {
        return this.damageSource
    }

    enum class DamageSource {
        ENTITY_ATTACK,
        FALL,
        VOID,
        PROJECTILE,
        DROWNING,
        CACTUS,
        LAVA,
        ON_FIRE,
        FIRE,
        ENTITY_EXPLODE,
        MAGIC_EFFECT,
        STARVE,
        API,
        COMMAND,
    }

}